src/main/scala/io/radicalbit/nsdb/connector/kafka/sink/NSDbSinkWriter.scala

```diff
@@ -148,14 +153,41 @@ class NSDbSinkWriter(connection: NSDB,
 
 object NSDbSinkWriter {
 
-  import io.radicalbit.nsdb.connector.kafka.sink.conf.Constants._
-
   private val logger = Logger(LoggerFactory.getLogger(classOf[NSDbSinkWriter]))
 
   private val defaultTimestampKeywords = Set("now", "now()", "sys_time", "sys_time()", "current_time", "current_time()")
 
   private def getFieldName(parent: Option[String], field: String) = parent.map(p => s"$p.$field").getOrElse(field)
 
+  private def policy(maxRetries: Int) = RetryPolicies.limitRetries(maxRetries - 1)
+
+  private def onFailure(throwable: Throwable, details: RetryDetails): Future[Unit] = {
+    logger.debug(throwable.getMessage)
+    Sleep[Future].sleep(1000.milliseconds)
+  }
+
+  private case class WriteFailureException(msg: String) extends RuntimeException(msg)
+
+  def writeWithDeliveryPolicy(semanticDelivery: Option[SemanticDelivery],
+                              fResult: => Future[List[RPCInsertResult]],
+                              retries: Option[Int]) = {
+    semanticDelivery match {
+      case Some(AtLeastOnce) =>
+        Await.result(
+          retryingOnAllErrors.apply(policy(retries.getOrElse(NSDbConfigs.NSDB_AT_LEAST_ONCE_RETRIES_DEFAULT)),
+                                    onFailure)(fResult.flatMap {
+            case rPCInsertResults if !rPCInsertResults.forall(_.completedSuccessfully) =>
+              Future.failed(WriteFailureException("Field 'completedSuccessfully' returns false"))
+            case rPCInsertResults =>
+              Future.successful(rPCInsertResults)
+          }),
+          10.seconds
```

Comment:
could we set this timeout as a config property? afaik there is already a PR in the making to address this point, hence we can merge this PR after the other one